pkg/virtctl/vm/restart.go

```diff
@@ -64,26 +64,23 @@ func (o *Command) restartRun(args []string, cmd *cobra.Command) error {
 		dryRunOption = []string{metav1.DryRunAll}
 		fmt.Printf("Dry Run execution\n")
 	}
+	
+	gracePeriodChanged := cmd.Flags().Changed(gracePeriodArg)
 
-	if cmd.Flags().Changed(gracePeriodArg) && !forceRestart {
-		return fmt.Errorf("Can not set gracePeriod without --force=true")
+	if gracePeriodChanged != forceRestart {
+		return fmt.Errorf("Must both use --force=true and set --grace-period.")
 	}
 
 	if forceRestart {
-		if cmd.Flags().Changed(gracePeriodArg) {
-			err = virtClient.VirtualMachine(namespace).ForceRestart(context.Background(), vmiName, &v1.RestartOptions{GracePeriodSeconds: &gracePeriod, DryRun: dryRunOption})
-			if err != nil {
-				return fmt.Errorf("Error restarting VirtualMachine, %v", err)
-			}
-		} else if !cmd.Flags().Changed(gracePeriodArg) {
-			return fmt.Errorf("Can not force restart without gracePeriod")
+		err = virtClient.VirtualMachine(namespace).ForceRestart(context.Background(), vmiName, &v1.RestartOptions{GracePeriodSeconds: &gracePeriod, DryRun: dryRunOption})
+		if err != nil {
+			return fmt.Errorf("Error force stopping VirtualMachine, %v", err)
 		}
 	} else {
 		err = virtClient.VirtualMachine(namespace).Restart(context.Background(), vmiName, &v1.RestartOptions{DryRun: dryRunOption})
 		if err != nil {
-			return fmt.Errorf("Error restarting VirtualMachine %v", err)
+			return fmt.Errorf("Error stopping VirtualMachine %v", err)
```

Comment:
why stopping? shouldn't it be `restarting`?

Reply:
That's correct, good catch. I've fixed it.